pkg/providers/instancetype/suite_test.go

```diff
@@ -1597,6 +1600,17 @@ var _ = Describe("InstanceType Provider", func() {
 				value := item.ValueFunc()
 
 				pod := coretest.UnschedulablePod(coretest.PodOptions{NodeSelector: map[string]string{item.Label: value}})
+				// Simulate multiple scheduling passes before final binding, this ensures that when real scheduling happens we won't
+				// end up with a new node for each scheduling attempt
```

Comment:
This looks like it would work ... I presume you can make this fail under "wrong" circumstances?

FWIW, there is a number of tests for in-flight nodes in core that may be helpful, like this one: https://github.com/kubernetes-sigs/karpenter/blob/34398e54f24b281619bf39e728480b422938e8b3/pkg/controllers/provisioning/scheduling/suite_test.go#L2412 (and others, search for "in-flight")

Reply:
Yes, I was able to reproduce the issue I was trying to reproduce with this pattern. Once I fixed, it fixed it here and also in the E2E test. This was really just a way to get the failure in a unit test rather than an E2E test since it was a bit easier to debug there.

It looks like they use a similar trick there (call `ExpectProvisionedNoBinding` a few times)